core/generator/generator.go

```diff
@@ -75,7 +75,16 @@ func Generate(
 		log.Fatal(ctx, log.KeyError, err)
 	}
 	if b != nil && (g.latestBlock == nil || b.Height == g.latestBlock.Height+1) {
-		s := state.Copy(g.latestSnapshot)
+		var s *state.Snapshot
+		if g.latestSnapshot == nil {
```

Comment:
That starts to make some functions look like parameter salad, but I gave it a try - WDYT?

Note that `Chain.ValidateBlockForSig` still needs to query the store for block 1 in order to instantiate a snapshot - or else have the initial block hash plumbed into it, too - which could be done easily if `Chain` included it as a member.